File: bench/h4fixtures/objc/main.m
// ObjC call-form fixture.
@interface Widget : NSObject
- (void)bump;
+ (Widget*)make;
@end
@implementation Widget
- (void)bump {}
+ (Widget*)make { return 0; }
@end

struct Ops { void (*init)(void); };
static void freeFn(void) {}

static void caller(struct Ops* ops)
{
    freeFn();                     // 1. C-style bare call
    ops->init();                  // 2. call through struct field
    Widget* w = [Widget make];    // 3. class message send
    [w bump];                     // 4. instance message send
}

File: bench/h4fixtures/c/main.c
/* C call-form fixture. */
struct Ops { int (*init)(void); };
static int freeFn(void) { return 1; }
#define MAC() freeFn()

static int caller(struct Ops* ops, struct Ops val)
{
    int a = freeFn();        /* 1. bare call */
    a += ops->init();        /* 2. arrow field call */
    a += val.init();         /* 3. dot field call */
    a += MAC();              /* 4. macro call */
    int (*fp)(void) = freeFn;
    a += fp();               /* 5. fn-pointer var call */
    return a;
}
